lib/idx/authenticate.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable complexity */
+import { AuthTransaction } from '../tx';
+import { OktaAuth, AuthorizeOptions, IdxTransactionMeta, RemediationValues } from '../types';
+import { interact } from './interact';
+import { introspect } from './introspect';
+import { remediate } from './remediate';
+
+export async function authenticate(authClient: OktaAuth, options: AuthorizeOptions): Promise<AuthTransaction> {
```

Comment:
the authentication flow uses `AuthorizeOptions`, looks a bit weird to me.

also, should we follow the signature from Laura's note:
```
function authenticate(sdk, { username, password }): Promise<TokenResponse>
```

Reply:
you mean in terms of returning tokens or returning a transaction?  If we return tokens we will have no choice but to throw an Exception for unsatified remediations. Returning the transaction is the current behavior of V1. I think it will be easier for us and customers to migrate if the interface hasn't changed too dramatically. I think we can make a V2 AuthTransaction that looks and acts very much like the V1 transaction. So existing code that says `res.status === SUCCESS` for example could continue to work.

Reply:
We can handle the compatibility issue in the upper-level wrapper method (signIn). For here, I think it's would be helpful to keep the API signature consistent even across different APIs. I think we can discuss more in the meeting to agree on a unified interface.